File: bcube-services/studio-service/src/main/java/com/bcube/studioservice/service/dto/request/CreateStudioRequest.java
package com.bcube.studioservice.service.dto.request;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class CreateStudioRequest {
    @NotNull(message = "Smartlock-ID ist erforderlich")
    private Long smartlockId;
    private String street;
    private int plz;
    private String city;
    private String country;
    private byte[] image;
    private List<byte[]> images;
    private String name;
    private String description;
    private String location;
}

File: bcube-web/src/app/models/requests/studio/create-studio-request.ts
export interface CreateStudioRequest {
    smartlockId: number;
    name: string;
    description: string;
    street: string;
    plz: number;
    city: string;
    country: string;
    image: number[];
    images: number[][];
    hourlyRateCents: number;
}
